NOTES.md

```diff
@@ -0,0 +1,95 @@
+# Notes
+
+## Requirements
+
+- Compact
+- Portable and human readable
+- Non-redundant*
+- Complete
+
+
+## Important terms of agreement
+
+- Every host that subscribes to the backend RESTful API is a free San Andreas Multiplayer service for San Andreas Multiplayer users.
+- Any user is free to choose what service they want to use.
+- The backend RESTful API needs to provide non-redundant*, complete and correct datasets regarding the requests it recieves.
+- Every subscription has a non-biased lifetime that is acceptible by the backend RESTful API.
+- Servers and the backend RESTful API can additionally agree on a custom subscription lifetime.
+- Multiple requests in a short period of time can be rate-limited by the backend RESTful API to prevent heavy workload.
+- Servers can't subscribe other servers.
+- The backend RESTful API can obtain IP addresses from the San Andreas Multiplayer API, and obtain server data accordingly.
+
+
+## Overview of interfaces
+
+- Servers can send data on an agreed format to the backend RESTful API.
+- Users can send requests on an agreed format to the backend RESTful API to recieve a list of subscriptions on an agreed format of San Andreas Multiplayer servers.
+
+## API
+
+### Server to backend RESTful API (Subscribe)
+
+The communication between the server and the backend RESTful API uses JSON (JavaScript Object Notation) that consists of these attributes:
+
+- Host: Domain name or IPv4 address
+- Hostname: Hostname of the server
+- Gamemode: Gamemode of the server
+- Language: Language of the server
+- MaxPlayers: Maximum amount of players that the server supports
+- Lifetime: Number of seconds to life as integer, -1 for default lifetime
+- Extended: This attribute is an object that cotains multiple attributes like for example website URL, contact email and etc.
+
+The message is contained in a HTTP GET request "/subscribe"
```

Comment:
I was planning on just using the `/server/{address}` endpoint with a POST method for this (just to simplify the amount of routes - both of these actions are getting and setting the same information so it makes sense to use the same endpoint). also "subscribe" usually means to request information periodically rather than to send or "announce" it.